packages/gatsby-transformer-cloudinary/create-image-node.js

```diff
@@ -49,24 +48,33 @@ exports.createImageNode = ({
     );
   }
 
+  const fingerprint = stringify({
+    cloudName,
+    height,
+    public_id,
+    breakpoints,
+    version,
+    width,
+  });
+
   const imageNode = {
     // These helper fields are only here so the resolvers have access to them.
     // They will *not* be available via Gatsby’s data layer.
-    cloudName: cloudName,
+    cloudName: cloudName || getPluginOptions().cloudName,
     public_id: public_id,
     version: version,
     originalHeight: height,
     originalWidth: width,
     breakpoints,
 
     // Add the required internal Gatsby node fields.
-    id: createNodeId(`CloudinaryAsset-${secure_url}`),
+    id: createNodeId(`CloudinaryAsset-${fingerprint}`),
```

Comment:
When creating a node based on the cloudName, publicId, width, and height, we don't have a secure_url to uniquely identify an image. Instead, we can use a stable stringification of cloudName, height, public_id, breakpoints, version, and width to create a unique identifier for each image.